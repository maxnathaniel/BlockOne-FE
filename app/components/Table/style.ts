import styled from '../../styles/styled-components';

interface ISidebarContainerProps {
  position: string;
  height: string;
  width: string;
}

export const TableContainer = styled.div`
  min-height: 400px;
  text-align: center;
`;

export const Header = styled.div`
  display: flex;
  border-bottom: 2px solid #E94F90;

  > p {
    flex: 1;
  }
`;

export const HeaderCell = styled.p`
  padding: 5px;
  color: #FFF;
  margin: 5px 0 !important;
`;

export const Row = styled.div`
  display: flex;

  > p {
    flex: 1;
  }
`;

export const Cell = styled.p`
  padding: 5px;
  color: #FFF;
  margin: 5px 0 !important;
  border-bottom: 1px solid #005f7b;
`;