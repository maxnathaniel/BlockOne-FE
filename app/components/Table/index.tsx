import React from 'react';
import numeral from 'numeral';
import { Cell, TableContainer, Header, HeaderCell, Row } from './style';

interface ITableProps  {
  data: any[];
}

const Table: React.FC<ITableProps> = (props) => {

  return (
    <TableContainer>
      <Header>
        <HeaderCell>Timestamp</HeaderCell>
        <HeaderCell>Action</HeaderCell>
        <HeaderCell>Description</HeaderCell>
        <HeaderCell>Amount</HeaderCell>
        <HeaderCell>Currency</HeaderCell>
      </Header>
      {
        props.data.length > 0 && props.data.map((row: any, index: number) =>
          (
            <Row key={index}>
              <Cell>{row.timestamp}</Cell>
              <Cell>{row.action}</Cell>
              <Cell>{row.description}</Cell>
              <Cell>{numeral(row.amount).format('$0,0.00')}</Cell>
              <Cell>{row.currency}</Cell>
            </Row>
          )
        )
      }


    </TableContainer>
  );
};

export default Table;