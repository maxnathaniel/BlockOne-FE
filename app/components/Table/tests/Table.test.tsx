import React from 'react';
import { render, getByText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Table from '../index';

const data = [
  { timestamp: new Date().toLocaleDateString(), action: 'Credit', description: 'N.A', amount: '10000', currency: 'SGD' }
];

describe('<Table />', () => {
  it('should render the various rows in Table', () => {
    const { container, getByText } = render(<Table data={data} />);

    expect(getByText('Credit')).toBeInTheDocument();
    expect(getByText('$10,000.00')).toBeInTheDocument();
    expect(getByText('N.A')).toBeInTheDocument();
    expect(getByText('SGD')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});