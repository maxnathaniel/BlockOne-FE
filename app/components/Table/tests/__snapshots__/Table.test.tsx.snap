// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render the various rows in Table 1`] = `
<div>
  <div
    class="sc-AykKC iLtsOi"
  >
    <div
      class="sc-AykKD dOvTbI"
    >
      <p
        class="sc-AykKE PnnHt"
      >
        Timestamp
      </p>
      <p
        class="sc-AykKE PnnHt"
      >
        Action
      </p>
      <p
        class="sc-AykKE PnnHt"
      >
        Description
      </p>
      <p
        class="sc-AykKE PnnHt"
      >
        Amount
      </p>
      <p
        class="sc-AykKE PnnHt"
      >
        Currency
      </p>
    </div>
    <div
      class="sc-AykKF TAKLS"
    >
      <p
        class="sc-AykKG dvNMrV"
      >
        2/17/2020
      </p>
      <p
        class="sc-AykKG dvNMrV"
      >
        Credit
      </p>
      <p
        class="sc-AykKG dvNMrV"
      >
        N.A
      </p>
      <p
        class="sc-AykKG dvNMrV"
      >
        $10,000.00
      </p>
      <p
        class="sc-AykKG dvNMrV"
      >
        SGD
      </p>
    </div>
  </div>
</div>
`;
